### DISCUSSION TITLE
用于 Chrome 的 PDF 翻译 的夜间模式

### DISCUSSION THREAD

**Discussion by yaoxiao6** [immersive-translate/immersive-translate#1670] [Ideas]
### 新功能 | New Feature

我希望在使用 Chrome 插件的 PDF 文档对照翻译的时候可以切换**夜间模式/黑暗模式**，即：_白底黑字_ 的 PDF 变成 _黑底白字_

### 为什么您想要添加这些功能 | Why you want to add these features

有时候我的工作环境光照不足，电脑屏幕太亮眼会不舒服，因此我会打开夜间模式。
系统可以自动切换，网页可以通过 `Dark Reader` 之类的插件做到，但是 PDF 文档翻译白花花的大底却无法变成黑色，会比较闪眼睛。希望提升改进一下。

### 提议 | Proposal

猜想1: 我假设：在 PDF 文档翻译之前，首先会对 PDF 文档内的文字图片做一次提取（之后才好作为 text 导入翻译引擎）。那么可以在提取文字之后，将原文和译文都输出为 HTML 而非 PDF。这样变成 HTML text 之后就很容易通过 CSS 改变颜色了。但是如何保持 PDF 原来的排版或许会是一个挑战

猜想2: 有没有什么比较强力的 CSS 可以覆盖所有 section 或者 div 的颜色配置？可以考虑利用那种 CSS 把 PDF 部分的颜色也改变掉。

**Comment by keyboard3** [comment]
技术上做不到，因为左侧是pdf.js框架通过canvas绘制的PDF原文，无法通过修改css来改变主题

**Reply by actforjason** [reply]
深色可以参考[doqment](https://addons.mozilla.org/zh-CN/firefox/addon/doqment/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search)吗
